functional-tests/src/test/groovy/pages/app/NotificationsPage.groovy

```diff
@@ -30,11 +32,12 @@ class NotificationsPage extends BaseAppPage {
    */
   List<Navigator> getSortedMatchingRows(String linkText) {
     waitFor {
-      notificationTable.$('.rt-tbody').children().has('.unread').has('.col-notification button', text: linkText)
-        .sort{
-          a, b ->
-            b.$('.col-credit-trade button').text().toInteger() <=> a.$('.col-credit-trade button').text().toInteger()
-        }
+      def tableRowsParent = notificationTable.$('.rt-tbody')
+      def unreadMatchingRows = tableRowsParent.children().has('.unread').has('.col-notification button', text: linkText)
+      def sortedUnreadMatchingRows = unreadMatchingRows.sort{ a, b ->
+        b.$('.col-credit-trade button').text().toInteger() <=> a.$('.col-credit-trade button').text().toInteger()
+      }
+      return sortedUnreadMatchingRows
```

Comment:
Broke the logic these 2 functions into individual steps, so it is easier to follow.